logs/stream/stream.go

```diff
@@ -44,41 +44,81 @@ import (
 func findReadyLogLines(logLines []state.LogLine, threshold time.Duration) ([]state.LogLine, []state.LogLine) {
 	var readyLogLines []state.LogLine
 	var tooFreshLogLines []state.LogLine
-	var lastPrimaryLogLine *state.LogLine
+	var lastLogLineWithPrefixPid int32 = -1
+	var lastMainLogLinePid int32 = -1
 
 	now := time.Now()
 
 	for _, logLine := range logLines {
+		ready := false
 		if logLine.LogLevel == pganalyze_collector.LogLineInformation_UNKNOWN {
```

Comment:
Maybe this is overthinking it, but I wonder if, for readability, we shouldn't invert the condition here and handle known-LogLevel lines first? What happens in the first branch depends on what happened to related lines in the second branch, so maybe having that other branch come first makes more sense?